### ISSUE TITLE
eval_acc param does not work with val_set param

### ISSUE THREAD

**Issue by rnyak** [IndicoDataSolutions/finetune#459]
**Describe the bug**
@benleetownsend @madisonmay 
I need to use an explicit validation set, so I use "val_set" param, but when I check TensorBoard, only training logs (loss, etc.) are displayed, but there is no info for val set. However, when I used "val_size" param, I can see val accuracy, loss, etc. results on TensorBoard.

Apparently, Classifier function does not use val set when we use `val_set` param instead of `val_size`, and, therefore, I believe best model is not saved based on validation accuracy. So, how can this issue be fixed? I need val_acc and loss results every certain steps when I use `val_set` param.

**Minimum Reproducible Example**
This is my finetuning command as an example:

```
"Classifier with single class for each input"
import time
start = time.time()
model = Classifier(n_epochs=1 , base_model=GPT2Model, tensorboard_folder ='/workspace/tensorboard', max_length=512, val_set =(valX, valY), chunk_long_sequences=False, keep_best_model= True, eval_acc = True)
model.fit(trainX, trainY)
print("total training time:", time.time() - start)
```

**Expected behavior**

**Additional context**

**Comment by benleetownsend** [comment]
I have had a quick look at this problem and I am able to reproduce. Behaviour seems to be correct in the case that you provide.
```
        val_set=(valX, valY),
        val_interval=100,
        val_size=len(valY)
```
Importantly here it is necessary to provide the val_size parameter that matches the length of the val dataset you are providing. Ideally this would be done internally. val_interval is the number of steps of train for every 1 validation run. Let me know if you have any problems with this approach and if it seems correct to you we can work at setting the val_size parameter internally.

**Comment by rnyak** [comment]
Thanks for your response @benleetownsend. I did the same thing, I set the val_size as the size of my Val dataset. But I still did no see "eval" results on the Tensorboard. I will test it again. 

I checked the source code in the input_pipeline.py file and to me the code should be modified. The train_test split  of the training set should be an option, and can be set as a flag. This is a problem if your training set includes oversampled records, and then your val set would have some of these records. That means training and val set samples might overlap, which won't be accurate. You probably would say, that's why we have "val_set" option for custom validation set,  but I am not 100% sure that it really uses my val_set, not the split from the training set when I also enter "val_size =len(valY)".  So, you might want to get rid of val_size when we  use val_set=(valX, valY), and val_size can be used with the "train_test_split =True" param.

Happy holidays!
